sharding-orchestration/sharding-orchestration-center/sharding-orchestration-center-apollo/src/main/java/org/apache/shardingsphere/orchestration/center/instance/ApolloInstance.java

```diff
@@ -17,177 +17,93 @@
     
     @Override
     public void init(final InstanceConfiguration config) {
-        initApolloConfig(config);
-        initApolloOpenApiClient();
-        initKeysRelationship();
-    }
-    
-    private void initApolloConfig(final InstanceConfiguration config) {
-        namespace = config.getNamespace();
-        appId = properties.getProperty("appId", "APOLLO_SHARDING_SPHERE");
-        env = properties.getProperty("env", "DEV");
-        clusterName = properties.getProperty("clusterName", ConfigConsts.CLUSTER_NAME_DEFAULT);
-        System.setProperty(APOLLO_KEY_APP_ID, appId);
-        System.setProperty(APOLLO_KEY_ENV, env);
-        System.setProperty(APOLLO_KEY_CLUSTER, clusterName);
-        System.setProperty(APOLLO_KEY_META, config.getServerLists());
-        apolloConfig = ConfigService.getConfig(namespace);
-    }
-    
-    private void initApolloOpenApiClient() {
-        administrator = properties.getProperty("administrator");
-        String apolloToken = properties.getProperty("token");
-        String portalUrl = properties.getProperty("portalUrl");
-        Integer connectTimeout = Ints.tryParse(properties.getProperty("connectTimeout"));
-        Integer readTimeout = Ints.tryParse(properties.getProperty("readTimeout"));
-        client = ApolloOpenApiClient.newBuilder().withPortalUrl(portalUrl)
-                .withConnectTimeout(connectTimeout == null ? ApolloOpenApiConstants.DEFAULT_CONNECT_TIMEOUT : connectTimeout)
-                .withReadTimeout(readTimeout == null ? ApolloOpenApiConstants.DEFAULT_READ_TIMEOUT : readTimeout)
-                .withToken(apolloToken).build();
-    }
-    
-    private void initKeysRelationship() {
-        List<OpenItemDTO> items = client.getNamespace(appId, env, clusterName, namespace).getItems();
-        Set<String> keys = Sets.newHashSet();
-        for (OpenItemDTO each : items) {
-            keys.add(each.getKey());
-        }
-        tree = ConfigTreeNode.create(keys, ".");
+        configWrapper = new ApolloConfigWrapper(config, properties);
+        openApiWrapper = new ApolloOpenApiWrapper(config, properties);
     }
     
     @Override
     public String get(final String key) {
-        return apolloConfig.getProperty(convertKey(key), "");
-    }
-    
-    private String convertKey(final String shardingSphereKey) {
-        return shardingSphereKey.replace(SHARDING_SPHERE_KEY_SEPARATOR, APOLLO_KEY_SEPARATOR).substring(1);
-    }
-    
-    private String deConvertKey(final String apolloKey) {
-        return new StringBuilder(SHARDING_SPHERE_KEY_ROOT).append(apolloKey.replace(APOLLO_KEY_SEPARATOR, SHARDING_SPHERE_KEY_SEPARATOR)).toString();
+        String value = configWrapper.getProperty(ConfigKeyUtils.path2Key(key));
+        return Strings.isNullOrEmpty(value) ? openApiWrapper.getValue(ConfigKeyUtils.path2Key(key)) : value;
     }
     
     @Override
     public List<String> getChildrenKeys(final String key) {
```

Comment:
I think maybe we could remove getChildrenKeys from config center interface.

Reply:
And close() method also could be discussed.

Reply:
The etcd does not integrate now, I think if it doesn't need getChildrenKeys, we can remove it.

close() method should be retain, but it is not need for all framework.